### PR TITLE
Pass metric argument into PixelSpotDetector

### PR DESCRIPTION
**Pull request by dganguli**
The PixelSpotDetector assigns each pixel in a stack (across C and R) to a barcode in the codebook via a distance function. Currently this distance function is hardcoded to be the euclidean distance. This PR generalizes the PixelSpotDetector to accept any distance function supported by sklearn.NearestNeighbors. 

FYI @kevinyamauchi

### CODE REVIEW

File: notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.ipynb [hunk 1]
```diff
@@ -394,7 +395,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.6.5"
+   "version": "3.6.3"
```
Comment (ambrosejcarr) [inline notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.ipynb]: @dganguli you should `brew upgrade python3` at some point. :)

File: starfish/pipeline/features/pixels/pixel_spot_detector.py [hunk 2]
```diff
@@ -9,19 +9,20 @@
...
 
 class PixelSpotDetector(PixelFinderAlgorithmBase):
-
     def __init__(
-            self, codebook: Codebook, distance_threshold: float,
-            magnitude_threshold: int, min_area: int, max_area: int, norm_order: int=2,
-            crop_x: int=0, crop_y: int=0, crop_z: int=0, **kwargs) -> None:
+            self, codebook: Codebook, metric: str, distance_threshold: float,
+            magnitude_threshold: int, min_area: int, max_area: int, norm_order: int = 2,
+            crop_x: int = 0, crop_y: int = 0, crop_z: int = 0, **kwargs) -> None:
```
Comment (ambrosejcarr) [inline starfish/pipeline/features/pixels/pixel_spot_detector.py]: I could have sworn that they _required_ there to be no spaces around the equals operator, but then I went and found these examples and now I don't know what to do!

https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html?highlight=default#functions

@ttung

Comment (dganguli) [reply starfish/pipeline/features/pixels/pixel_spot_detector.py]: this is the result of running Pycharm's "Reformat Code" tool, which adheres to whatever PEP-8y thing Pycharm defaults to. I can change the defaults if you'd like
